day_5_supply_stacks/stack_parser.rb

```diff
@@ -0,0 +1,78 @@
+require 'minitest/autorun'
+
+class StackParser
+  def initialize(stack_input)
+    @stack_input = stack_input
+    @stack_rows = stack_rows
```

Comment:
As discussed previously, this is kinda weird. When I started to parse the code, my immediate reaction was 'where's `stack_rows` coming from, and why is it being set up here.

A better solution would be memoization imo.

So:

1) remove that line 
2) change stack_rows to

```ruby
  def stack_rows
    @_stack_rows ||= stack_input.map do |row|
      row.each_char.each_slice(4).map do |crate|
        crate.join.tr("[]", "").strip
      end
    end
  end
```

Now you have the same code exactly (basically what you did here is 'manual  memoization' (not sure that's a name, maybe) - computed the value once, and stored it in a variable so that you don't have to compute it again).

'Normal' memoization does the same in a different way - it stores the value in a variable once, and after that you can call the method as many times as you want, it returns the stored value.

Reply:
Good point. This felt definitely weird.